### PR TITLE
Create a table of contents for the migration guide

### LINKED ISSUE #5126: Create a table of contents for the migration guide

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Fixes #5126

### WHAT is this pull request doing?

- Adds a table of contents to the migration guide

### CODE REVIEW

**Comment by lgriffee** [comment]
Using the _Markdown All in One_ VSCode plugin I was able to generate an automatic table of contents (which will update automatically if we add more headers and you have the plugin)! I excluded the primary header and the table of contents header. Is there anything else we want to exclude? I'm debating whether having all the functions and mixins listed is over-the-top or helpful...

File: documentation/guides/migrating-from-v8-to-v9.md [hunk 1]
```diff
@@ -1,7 +1,53 @@
-# Migrating from v8 to v9
+# Migrating from v8 to v9 <!-- omit in toc -->
```
Comment (lgriffee) [inline documentation/guides/migrating-from-v8-to-v9.md]: `<!-- omit in toc -->` comment to exclude the header from the automatic toc generation by _Markdown All in One_ VSCode plugin.

File: documentation/guides/migrating-from-v8-to-v9.md [hunk 2]
```diff
@@ -1,7 +1,53 @@
...
+    - [`skeleton-page-header-layout()`](#skeleton-page-header-layout)
+    - [`skeleton-page-secondary-actions-layout()`](#skeleton-page-secondary-actions-layout)
+    - [`skeleton-shimmer()`](#skeleton-shimmer)
+    - [`spacing()`](#spacing)
+    - [`state()`](#state)
+    - [`unstyled-link()`](#unstyled-link)
+    - [`unstyled-list()`](#unstyled-list)
+    - [`when-not-printing()`](#when-not-printing)
+    - [`when-printing()`](#when-printing)
+    - [`z-index()`](#z-index)
```
Comment (lgriffee) [inline documentation/guides/migrating-from-v8-to-v9.md:46]: Over-the-top? Helpful?

File: documentation/guides/migrating-from-v8-to-v9.md (hunk 1 above)
Comment (alex-page) [reply documentation/guides/migrating-from-v8-to-v9.md]: Unless the Markdown All in One plugin is used by all contributors we should probably remove the comments.

**Review by alex-page — APPROVED** [review]
I think this is good and the sub headings are helpful. We can always iterate quickly if we get feedback.

File: documentation/guides/migrating-from-v8-to-v9.md (hunk 2 above)
Comment (aveline) [reply documentation/guides/migrating-from-v8-to-v9.md:46]: I think it's helpful, let's keep for now.

**Review by aveline — APPROVED** [review]

**Review by sam-b-rose — APPROVED** [review]

**Pull request merged**